File: src/main/java/app/tweditor/DatabaseUpdateListener.java
package app.tweditor;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;

public class DatabaseUpdateListener
  implements ActionListener, DocumentListener
{
  private final GameSession session;

  public DatabaseUpdateListener(GameSession session)
  {
    this.session = session;
  }

  public void actionPerformed(ActionEvent ae)
  {
    if ((this.session.getDatabase() != null) && (!this.session.isDataChanging())) {
      this.session.setDataModified(true);
      Main.mainWindow.setTitle(null);
    }
  }

  public void changedUpdate(DocumentEvent de)
  {
    if ((this.session.getDatabase() != null) && (!this.session.isDataChanging())) {
      this.session.setDataModified(true);
      Main.mainWindow.setTitle(null);
    }
  }

  public void insertUpdate(DocumentEvent de)
  {
    if ((this.session.getDatabase() != null) && (!this.session.isDataChanging())) {
      this.session.setDataModified(true);
      Main.mainWindow.setTitle(null);
    }
  }

  public void removeUpdate(DocumentEvent de)
  {
    if ((this.session.getDatabase() != null) && (!this.session.isDataChanging())) {
      this.session.setDataModified(true);
      Main.mainWindow.setTitle(null);
    }
  }
}

File: src/main/kotlin/app/tweditor/DatabaseUpdateListener.kt
package app.tweditor

import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

class DatabaseUpdateListener(private val session: GameSession) : ActionListener, DocumentListener {
    override fun actionPerformed(ae: ActionEvent?) {
        if (session.database != null && !session.isDataChanging()) {
            session.setDataModified(true)
            Main.mainWindow!!.setTitle(null)
        }
    }

    override fun changedUpdate(de: DocumentEvent?) {
        if (session.database != null && !session.isDataChanging()) {
            session.setDataModified(true)
            Main.mainWindow!!.setTitle(null)
        }
    }

    override fun insertUpdate(de: DocumentEvent?) {
        if (session.database != null && !session.isDataChanging()) {
            session.setDataModified(true)
            Main.mainWindow!!.setTitle(null)
        }
    }

    override fun removeUpdate(de: DocumentEvent?) {
        if (session.database != null && !session.isDataChanging()) {
            session.setDataModified(true)
            Main.mainWindow!!.setTitle(null)
        }
    }
}
